src/util/integer_gmp_imp.cpp

```diff
@@ -461,8 +461,10 @@ int64_t Integer::getSigned64() const
       return getLong();
     }
     // ensure there is no overflow.
-    // mpz_sizeinbase ignores the sign bit, thus at most 63 bits.
-    CheckArgument(mpz_sizeinbase(d_value.get_mpz_t(), 2) < 64,
+    CheckArgument(mpz_class(std::to_string(std::numeric_limits<int64_t>::min()))
+                          <= d_value
+                      && d_value <= mpz_class(std::to_string(
+                             std::numeric_limits<int64_t>::max())),
                   this,
                   "Overflow detected in Integer::getSigned64().");
     return std::stoll(toString());
```

Comment:
Can't we rely on `std::stoll` here to throw an `std::out_of_range` exception instead of using this check?